internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -86,14 +87,17 @@ func (t *Translator) ProcessBackendTrafficPolicies(backendTrafficPolicies []*egv
 			res = append(res, policy)
 
 			// Negative statuses have already been assigned so its safe to skip
-			route := resolveBTPolicyRouteTargetRef(policy, routeMap)
-			if route == nil {
+			route, resolveErr := resolveBTPolicyRouteTargetRef(policy, routeMap)
+			if route == nil && resolveErr == nil {
```

Comment:
is this so that you can update `status.SetResolveErrorForPolicyAncestors` by first computing ``ancestorRefs` ?

Reply:
yes. in the previous commit, the `ResolveError` is calculated in each resolve method and then got surfaced, which requires `ancestorRefs` to be computing again, since the computing logic are the same. so by returning this `resolveErr`, we just need to compute `ancestorRefs` once.

Reply:
thoughts on simplifying this logic ?
```
if route == nil {
  continue
}

ancestorRefs := getBTPAncestorReferences()

if resolveErr != nil {
  status.SetResolveErrorForPolicyAncestors
  continue
}
```

Reply:
yes, we can simplify the first if statement

Reply:
as for the get btpAncestorRefs part, since this logic can be reused by other policies which can target to xroute, seems better to wrap it up into a more generic method.

Reply:
> as for the get btpAncestorRefs part, since this logic can be reused by other policies which can target to xroute, seems better to wrap it up into a more generic method.

actually this loop is also used to collect `gatewayRouteMap`, not worthy to introduce another loop. so just done some simplification